sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockEventProcessorHandlersTests.cs

```diff
@@ -0,0 +1,244 @@
+using Azure.Storage.Blobs;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    /// <summary>
+    ///   The suite of tests defining the snippets used in the Event Hubs
+    ///   Sample07_MockEventProcessorHandlersTests sample.
+    /// </summary>
+    ///
+    [TestFixture]
+    [Category(TestCategory.Live)]
+    [Category(TestCategory.DisallowVisualStudioLiveUnitTesting)]
+    public class Sample07_MockEventProcessorHandlersTests
+    {
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProcessEventHandlerCheckpoint()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_ProcessEventHandlerCheckpoint
+
+            // Application handler checkpoints after processing defined number of events.
+
+            const int EventsBeforeCheckpoint = 2;
+            var partitionEventCount = new ConcurrentDictionary<string, int>();
+            bool checkpointUpdated = false;
+
+            async Task processEventHandler(ProcessEventArgs args)
+            {
+                try
+                {
+                    checkpointUpdated = false;
+
+                    await Application.ProcessEventAsync(args);
+
+                    string partition = args.Partition.PartitionId;
+
+                    int eventsSinceLastCheckpoint = partitionEventCount.AddOrUpdate(
+                        key: partition,
+                        addValue: 1,
+                        updateValueFactory: (_, currentCount) => currentCount + 1);
+
+                    if (eventsSinceLastCheckpoint >= EventsBeforeCheckpoint)
+                    {
+                        await args.UpdateCheckpointAsync();
+                        partitionEventCount[partition] = 0;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Application.HandleProcessingException(args, ex);
+                }
+            }
+
+            // Initialize the args and simulate checkpoint update method.
+
+            var partitionId = "0";
+            var eventBody = new BinaryData("This is a sample event body");
+            var eventData = EventHubsModelFactory.EventData(eventBody);
+            var partitionContext = EventHubsModelFactory.PartitionContext(partitionId);
+            var eventArgs = new ProcessEventArgs(partitionContext, eventData, _ => { checkpointUpdated = true; return Task.CompletedTask; });
+
+            for (int i = 1; i <= EventsBeforeCheckpoint; i++)
+            {
+                // Execute the handler and validate that the event handling has been checkpointed.
+
+                await processEventHandler(eventArgs);
+
+                if (i % EventsBeforeCheckpoint == 0)
+                    Assert.IsTrue(checkpointUpdated);
+                else
+                    Assert.IsFalse(checkpointUpdated);
```

Comment:
A comment about the if condition may be helpful for those not familiar with mod.

```suggestion
                if (i % EventsBeforeCheckpoint == 0)
                {
                    Assert.That(checkpointUpdated, Is.True);
                }
                else
                {
                    Assert.That(checkpointUpdated, Is.False);
                }
```

Friends don't let friends skip curly braces.   We're not monsters, after all.

Reply:
Could also  streamline a bit, if you wanted:

```csharp

var asertCondition = i switch
{
    _ when (i % EventsBeforeCheckpoint == 0) => Is.True,
    _ => Is.False
};

Assert.That(checkpointUpdated, asertCondition );
```